src/cmd-compress

```diff
@@ -150,10 +164,98 @@ def compress_one_builddir(builddir):
     return at_least_one
 
 
-changed = []
-for arch in builds.get_build_arches(build):
-    builddir = builds.get_build_dir(build, arch)
-    changed.append(compress_one_builddir(builddir))
+def uncompress_one_builddir(builddir):
+    # heavily based on the compress_one_builddir
+    print(f"Uncompressing: {builddir}")
+    buildmeta_path = os.path.join(builddir, 'meta.json')
+    with open(buildmeta_path) as f:
+        buildmeta = json.load(f)
 
-if not any(changed):
-    print("All builds already compressed")
+    tmpdir = 'tmp/uncompress'
+    if os.path.isdir(tmpdir):
+        shutil.rmtree(tmpdir)
+    os.mkdir(tmpdir)
+
+    # Note we mutate the build dir in place, similarly to the buildextend
+    # commands.  One cool approach here might be to `cp -al` the whole build
+    # dir, mutate it, then RENAME_EXCHANGE the two... though it doesn't seem
+    # Python exposes it yet so that'd require some hacking around. For now, we
+    # just guarantee that `compress` is idempotent and can resume from
+    # failures.
+
+    at_least_one = False
+
+    only_artifacts = None
+    if len(args.artifact) > 0:
+        only_artifacts = set(args.artifact)
+
+    for img_format, img in buildmeta['images'].items():
+        if img.get('skip-compression', False):
+            print(f"Skipping {img_format}")
+            continue
+        if only_artifacts is not None and img_format not in only_artifacts:
+            continue
+
+        file = img['path']
+        filepath = os.path.join(builddir, file)
+        # uncompress only compressed files
+        if file.endswith(tuple(ext_dict.values())):
+            tmpfile = os.path.join(tmpdir, strip_ext(file))
+            # SHA256 + size for uncompressed image was already calculated
+            # during 'build'
+            with open(tmpfile, 'wb') as f:
+                if file.endswith('xz'):
+                    t = xz_threads()
+                    run_verbose(['xz', '-dc', f'-T{t}', filepath], stdout=f)
+                elif file.endswith('gz'):
+                    run_verbose(['gzip', '-dc', filepath], stdout=f)
+                else:
+                    print(f"Unknown sufix of file {file}")
+            file_without_ext = strip_ext(file)
+            filepath_without_ext = strip_ext(filepath)
+            uncompressed_size = os.path.getsize(tmpfile)
+            uncompressed_sha256 = sha256sum_file(tmpfile)
+            # Check integrity of the uncompressed images
+            if uncompressed_sha256 != img['uncompressed-sha256']:
+                print(f"sha256 missmatch for {filepath_without_ext}")
+                print(f"{uncompressed_sha256}!={img['uncompressed-sha256']}")
+            if uncompressed_size != img['uncompressed-size']:
+                print("size missmatch for {filepath_without_ext}")
+                print(f"{uncompressed_size}!={img['size']}")
+
+            img['uncompressed-sha256'] = None
+            img['uncompressed-size'] = None
```

Comment:
```suggestion
            del img['uncompressed-sha256']
            del img['uncompressed-size']
```

This leaves us with null entries in the meta.json at the end. We want
them to not exist at all so let's delete them from the dictionary.